### PR TITLE
Admission api

### PR DESCRIPTION
**Pull request by sathyaprabaS**

### CODE REVIEW

File: controllers/admin/galleryController.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * @param {Response} res - The Express response object
+ */
+
+exports.createGallery = async (req, res, next) => {
+  var uploadedGalleryImages = [];
+  try {
+    const formData = req.body;
+    var galleryImageS3Location = req.files;
+
+      if (req.files) {
```
[the remark this answers is not in the document]
Comment (sathyaprabaS) [reply controllers/admin/galleryController.js:19]: length checked

File: controllers/admin/galleryController.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * @param {Request} req - The Express request object
+ * @param {Response} res - The Express response object
+ */
+
+exports.createGallery = async (req, res, next) => {
+  var uploadedGalleryImages = [];
+  try {
+    const formData = req.body;
+    var galleryImageS3Location = req.files;
```
[the remark this answers is not in the document]
Comment (sathyaprabaS) [reply controllers/admin/galleryController.js:17]: changed

File: controllers/api/admissionController.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * @param {Request} req - The Express request object
+ * @param {Response} res - The Express response object
+ */
+
+exports.createAdmission = async (req, res, next) => {
+  try {
+      const admissionData = req.body;
+
+    const Admission = {
```
[the remark this answers is not in the document]
Comment (sathyaprabaS) [reply controllers/api/admissionController.js:16]: changed

File: controllers/api/admissionController.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      religion: admissionData.religion.trim(),
+      fatherName: admissionData.fatherName.trim(),
+      motherName: admissionData.motherName.trim(),
+      address: admissionData.address.trim(),
+      pnoneNumber: admissionData.pnoneNumber,
+    };
+    var newAdmission = await AdmissionModel.create(Admission);
+    res.json(newAdmission);
+  } catch (error) {
+    console.error(error);
```
[the remark this answers is not in the document]
Comment (sathyaprabaS) [reply controllers/api/admissionController.js:33]: error throwed

File: controllers/api/galleryController.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @param {Request} req - The Express request object
+ * @param {Response} res - The Express response object
+ */
+
+
+exports.getAllGallery = async (req, res, next) => {
+  try {
+    const galleryImages = await GalleryModel.find();
+    if (galleryImages.length === 0) {
```
[the remark this answers is not in the document]
Comment (sathyaprabaS) [reply controllers/api/galleryController.js:16]: condition removed

File: controllers/admin/admissionController.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+const AdmissionModel = require("../../database/models/admission");
+/**
+ * @param {Request} req - The Express request object
+ * @param {Response} res - The Express response object
+ */
+
+
+exports.getAllAdmission = async (req, res, next) => {
+  try {
+    const Admissions = await AdmissionModel.find();
```
Comment (praveensumathi) [inline controllers/admin/admissionController.js:15]: change to camelcase

File: controllers/admin/galleryController.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      const { galleryIds } = req.body;
+
+      const gallery = await GalleryModel.find(
+        { _id: { $in: galleryIds } },
+        { imageUrl: 1 }
+      );
+
+      var deleteImages = [];
+      for (const item of gallery) {
+        if (item.imageUrl && item.imageUrl.length > 0) {
```
[the remark this answers is not in the document]
Comment (sathyaprabaS) [reply controllers/admin/galleryController.js:56]: length check removed

File: controllers/admin/galleryController.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        const imageUrl = await uploadToS3(
+          file.buffer,
+          file.originalname,
+          file.mimetype
+        );
+        uploadedGalleryImages.push(imageUrl);
+      }
+    }
+
+    if (uploadedGalleryImages.length === 0) {
```
[the remark this answers is not in the document]
Comment (sathyaprabaS) [reply controllers/admin/galleryController.js:30]: changed

File: controllers/admin/admissionController.js (hunk 6 above)
Comment (sathyaprabaS) [reply controllers/admin/admissionController.js:15]: changed

**Review by praveensumathi — APPROVED** [review]

**Pull request merged**